tests/php/test-amp-beta-tester.php

```diff
@@ -0,0 +1,528 @@
+	}
+
+	/**
+	 * Test show_unstable_reminder().
+	 *
+	 * @covers \AMP_Beta_Tester\show_unstable_reminder()
+	 */
+	public function test_show_unstable_reminder() {
+		if ( ! \AMP_Beta_Tester\is_pre_release( AMP__VERSION ) ) {
+			$this->markTestSkipped( 'Unstable reminder is only shown on non-stable versions.' );
+		}
+
+		global $show_admin_bar, $wp_admin_bar;
+
+		$show_admin_bar = true;
+
+		_wp_admin_bar_init();
+
+		AMP_Beta_Tester\show_unstable_reminder( $wp_admin_bar );
+
+		$node = $wp_admin_bar->get_node( 'amp-version-code' );
+
+		$this->assertEquals( sprintf( 'v%s', AMP__VERSION ), $node->title );
+		$this->assertEquals( admin_url( 'plugins.php?s=amp&plugin_status=active' ), $node->href );
+	}
+
+	/**
+	 * Mock a failed HTTP request to GitHub's API.
+	 *
+	 * @param mixed  $preempt Whether to preempt an HTTP request's return value. Default false.
+	 * @param mixed  $r       HTTP request arguments.
+	 * @param string $url     The request URL.
+	 * @return WP_Error Simulated HTTP error.
+	 */
+	public function mock_http_request_error( $preempt, $r, $url ) {
+		if ( false === strpos( $url, 'api.github.com' ) ) {
+			return $preempt;
+		}
+
+		return new WP_Error( 'failed_http_error' );
+	}
+
+	/**
+	 * Mock a successful HTTP request to GitHub's API.
+	 *
+	 * @param mixed  $preempt Whether to preempt an HTTP request's return value. Default false.
+	 * @param mixed  $r       HTTP request arguments.
+	 * @param string $url     The request URL.
+	 * @return array Response data.
+	 */
+	public function mock_github_api_http_request( $preempt, $r, $url ) {
+		if ( false === strpos( $url, 'api.github.com' ) ) {
+			return $preempt;
+		}
+
+		$body = empty( $this->custom_github_api_response )
+			? $this->generate_github_api_response()
+			: $this->custom_github_api_response;
+
+		return [
+			'body'          => $body,
+			'headers'       => [],
+			'response'      => [
+				'code'    => 200,
+				'message' => 'ok',
+			],
+			'cookies'       => [],
+			'http_response' => null,
+		];
+	}
+
+	/**
+	 * Generate a custom GitHub API response.
+	 *
+	 * @param string $version         AMP version to use.
+	 * @param bool   $is_pre_released Whether to set release as pre-release or not.
+	 * @param bool   $with_zip        Whether to include an 'amp.zip' asset or not.
+	 * @return string
+	 */
+	private function generate_github_api_response( $version = '1.0.0-beta', $is_pre_released = true, $with_zip = true ) {
+		$response = [
+			'url'              => 'https://api.github.com/repos/ampproject/amp-wp/releases/20819569',
+			'assets_url'       => 'https://api.github.com/repos/ampproject/amp-wp/releases/20819569/assets',
+			'upload_url'       => 'https://uploads.github.com/repos/ampproject/amp-wp/releases/20819569/assets{?name,label}',
+			'html_url'         => "https://github.com/ampproject/amp-wp/releases/tag/${version}",
+			'id'               => 20819569,
+			'node_id'          => 'MDc6UmVsZWFzZTIwODE5NTY5',
+			'tag_name'         => "${version}",
```

Comment:
An actual cast will better communicate the intention here.
```suggestion
			'tag_name'         => (string) $version,
```